src/lib/ArrayInfo.ts

```diff
@@ -1,121 +1,179 @@
-import {DataError} from "./Errors"
-import {KeyValue} from "./Utils"
+import { DataError } from './Errors'
+import { KeyValue } from './Utils'
 
 function isInt(value: any) {
-    return !isNaN(value) &&
-        Number(value) == value && !isNaN(parseInt(value, 10))
+  return !isNaN(value) && Number(value) == value && !isNaN(parseInt(value, 10))
 }
 
-export const arrayRegex = () => /^([\.0-9a-zA-Z_$\-][0-9a-zA-Z_\-$\.]*)\[((?!(\]|\[)).*|)\]$/gm
+export const arrayRegex = () => /^([\.0-9a-zA-Z_$\-][0-9a-zA-Z_\-$\.]*)(.*)/gm
+export const arrayIndiciesRegex = /\[(.*?)\]/g
+export const malformedArrayRegex = /\]|\[/g
 const regexCache = {} as KeyValue
 
 export class ArrayInfo {
-    readonly property: string
-    readonly index: number = 0
-    readonly append: boolean = false
-
-
-    constructor(property: string, index: any) {
-        this.property = property
-        this.append = index === ""
-        if (isInt(index)) {
-            this.index = parseInt(index)
-        } else if (!this.append) {
-            throw new DataError("Only numerical values accepted for array index", 200)
-        }
+  readonly property: string
+  readonly index: number = 0
+  readonly append: boolean = false
+  readonly indicies = [] as any[]
+
+  constructor(property: string, indicies: any[]) {
+    this.property = property
+    const index = indicies[0] ?? 0
+    this.append = index === '' || indicies[indicies.length - 1] === ''
+    this.indicies = indicies
+    if (isInt(index)) {
+      this.index = parseInt(index)
+    } else if (!this.append) {
+      throw new DataError('Only numerical values accepted for array index', 200)
+    }
+  }
+
+  /**
+   * Check if the property want to access an Array
+   * @param property
+   */
+  public static processArray(property?: string): ArrayInfo | null {
+    if (typeof property === 'undefined') {
+      return null
     }
 
-    /**
-     * Check if the property want to access an Array
-     * @param property
-     */
-    public static processArray(property?: string): ArrayInfo | null {
-        if (typeof property === 'undefined') {
-            return null
-        }
-
-        if (regexCache[property]) {
-            return regexCache[property]
-        }
-
-        const arrayIndexRegex = arrayRegex()
-        const match = arrayIndexRegex.exec(property.trim())
-        if (match != null) {
-            return (regexCache[property] = new ArrayInfo(match[1], match[2]))
-        }
-        return null
+    if (regexCache[property]) {
+      return regexCache[property]
     }
 
-    /**
-     * Get the index for the array
-     * @param data
-     * @param avoidProperty
-     */
-    public getIndex(data: KeyValue, avoidProperty?: boolean): number {
-        if (avoidProperty === undefined) {
-            avoidProperty = false
+    const arrayIndexRegex = arrayRegex()
+    const match = arrayIndexRegex.exec(property.trim())
+    if (match != null && match[2]) {
```

Comment:
Commenting the code would be nice to understand what each matches represent.

Maybe we should use [Named capturing groups](https://stackoverflow.com/questions/5367369/named-capturing-groups-in-javascript-regex) also in the regex

Reply:
Removed the previous regex entries in lieu of using easily understood functions.